modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/MappingServiceImpl.java

```diff
@@ -169,6 +223,8 @@ private CompletableFuture<List<MappedFragment>> map0(MultiStepPlan multiStepPlan
                             }
 
                             lastAttemptSucceed = true;
+
+                            break;
```

Comment:
good catch. I think potentially more obvious will be add the condition to for loop instead of `break` statement here

Reply:
Done, thanks